flight/PiOS/Common/pios_rcvr.c

```diff
@@ -108,6 +120,28 @@ int32_t PIOS_RCVR_Read(uintptr_t rcvr_id, uint8_t channel)
   return rcvr_dev->driver->read(rcvr_dev->lower_id, channel);
 }
 
+#define MIN_WAKE_INTERVAL_MS 4	/* 250Hz ought to be enough for anyone*/
+
+bool PIOS_RCVR_WaitActivity(uint32_t timeout_ms) {
+  return PIOS_Semaphore_Take(rcvr_activity, timeout_ms);
+}
+
+void PIOS_RCVR_Active() {
+  if (PIOS_DELAY_DiffuS(rcvr_last_wake) >= MIN_WAKE_INTERVAL_MS) {
+    rcvr_last_wake = PIOS_DELAY_GetRaw();
+    PIOS_Semaphore_Give(rcvr_activity);
+  }
+}
+
+void PIOS_RCVR_ActiveFromISR() {
+  if (PIOS_DELAY_DiffuS(rcvr_last_wake) >= MIN_WAKE_INTERVAL_MS) {
+    bool dont_care;
```

Comment:
See this other brain-dead line though: https://github.com/d-ronin/dRonin/blob/779741aacf1766e03bded34458a21e990fdcfa61/flight/PiOS/Common/pios_semaphore.c#L155